### ISSUE TITLE
[Input Text] Proposal to add custom props to the suggestions button

### ISSUE THREAD

**Issue by rodrigoea** [grommet/grommet#6115]
### Expected Behavior

I want to propose adding the ability to pass props to the `suggestions` button that exists in the `InputText` component. I wanted to add some attributes to the button or even be able to customize the button based on my existing custom theme for example. I could do it with javascript but wouldn't be the best way :)

### Actual Behavior

As far as I saw, currently, it's not possible to customize the suggestions button props in the `InputText` component:

https://github.com/grommet/grommet/blob/316e802ed2a05f71344977f6b3eb0f7283f30e5d/src/js/components/TextInput/TextInput.js#L370-L391

### URL, screen shot, or Codepen exhibiting the issue

<img width="977" alt="image" src="https://user-images.githubusercontent.com/1905961/167046920-9e5a4006-d7c0-402f-a257-9a4bf6dcb1e6.png">

---

Does that make sense? If so, I'm happy to open a PR doing that 🚀

**Comment by ericsoderberghp** [comment]
Can you share a screenshot or sketch of what you are aiming for visually?

**Comment by rodrigoea** [comment]
@ericsoderberghp what I mainly want to achieve is the ability to add some custom `data-attrs` to the button element.

In this Codesandbox you see that what we currently can do is to add a `Box` in the suggestion label with a `data-custom-attr`, but what I wanted is to add this custom attribute in the button.

<img width="854" alt="image" src="https://user-images.githubusercontent.com/1905961/167048764-9fc47711-9841-4df3-b068-78e38a978387.png">

I think if we just add a prop like `suggestionsProps` like Grommet already has for `dropProps`, we'd be able to customize the button props by maybe simply destructuring the prop?

**Comment by rodrigoea** [comment]
<img width="1657" alt="image" src="https://user-images.githubusercontent.com/1905961/167049249-94d8633d-e698-4d7b-bca4-3150dcc7b68b.png">

**Comment by rodrigoea** [comment]
it would be something like this: https://github.com/rodrigoea/grommet/commit/d345f8be4b03f3e96f5e0655f3652edd2803c534

**Comment by jcfilben** [comment]
We do allow custom suggestions, see [this storybook example](https://storybook.grommet.io/?path=/story/input-textinput-custom-themed-custom-suggestions--custom-suggestions&globals=theme:base). However, I realize this method doesn't let you edit the actual button but instead allows you to edit the content of the button. Can you share what you are trying to achieve that cannot be done without editing the button? I'm just trying to better understand the use case

**Comment by rodrigoea** [comment]
@jcfilben thanks for your reply!

I use a tool that collects data for click events, and I want to track clicks in each item of the suggestion list by passing a custom data attribute to the buttons.

I managed to work around it by passing a custom label (similar to what you shared), this way the tool can capture the clicks 👍🏻  - It works for now!

Since I managed to work around it, I'm closing the issue 😄  - thanks!
